daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -2378,7 +2414,7 @@ case class FNCountExpr(nameAsParsed: String, fnQName: RefQName, args: List[Expre
 }
 
 case class FNExactlyOneExpr(nameAsParsed: String, fnQName: RefQName, args: List[Expression])
-  extends FunctionCallArrayBase(nameAsParsed, fnQName, args) {
+  extends FunctionCallArrayOrOptionalBase(nameAsParsed, fnQName, args) {
 
```

Comment:
Can we update this compiledDPath in this class to also call `checkArgArrayOrOptional(0)`? We don't currently supported it, but if we do add support it would be good to already have this check.

Reply:
This call to checkArgArrayOrOptional(0) now made by FNExactlyOneExpr.